android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java

```diff
@@ -537,6 +553,56 @@ private Cursor getCursorForPhone(String phone) {
     return null;
   }
 
+  private Cursor getCursorForEmailOrName(String query, boolean queryName) {
```

Comment:
For me, it hard to clear your idea, srr.

Reply:
btw, IMO, it will help you a lot: https://github.com/linagora/linshare-mobile-android-app/blob/master/data/src/main/java/com/linagora/android/linshare/data/datasource/contact/DeviceContactDataSource.kt